web/api/v1/api.go

```diff
@@ -889,6 +893,50 @@ func (api *API) serveFlags(r *http.Request) apiFuncResult {
 	return apiFuncResult{api.flagsMap, nil, nil, nil}
 }
 
+// Stat holdes the information about individual cardinality.
+type stat struct {
+	Name  string `json:"name"`
+	Value uint64 `json:"value"`
+}
+
+// TSDBStatus has information of cardinality statastics from postings.
+type tsdbStatus struct {
+	QueryDuration                    string `json:"queryTime"`
+	ScrapeValueCountByMetricsName    []stat `json:"scrapeValueCountByMetricsName"`
+	LabelValueCountByLabelName       []stat `json:"labelValueCountByLabelName"`
+	MemoryUsageInBytesByLabelName    []stat `json:"memoryUsageInBytesByLabelName"`
+	ScrapeValueCountByLabelValuePair []stat `json:"scrapeValueCountByLabelValuePair"`
+}
+
+func (api *API) serveTSDBStatus(r *http.Request) apiFuncResult {
+	db := api.db()
+	if db == nil {
+		return apiFuncResult{nil, &apiError{errorUnavailable, errors.New("TSDB not ready")}, nil, nil}
+	}
+	conver := func(stats []index.Stat) []stat {
+		result := make([]stat, 0, len(stats))
+		for _, item := range stats {
+			item := stat{Name: item.Name, Value: item.Count}
+			result = append(result, item)
+		}
+		return result
+	}
+
+	startTime := time.Now().UnixNano()
```

Comment:
I would only do the conversion from `time.Duration` at the last possible moment, so just `time.Now()` here, and then below you can do `time.Since(startTime).Seconds()` and get the elapsed duration in seconds without all the math.